CHANGELOG.md

```diff
@@ -2,6 +2,42 @@
 All notable changes to this project will be documented in this file.
 This project adheres to [Semantic Versioning](http://semver.org/).
 
+## Unreleased
+### Added
+- added `prebuilt_docker_image_task_types`. These are the task types that allow non-artifact docker images; if `None`, all task types are allowed.
+- added `get_in_tree_template`, `get_action_context_and_template`, `get_jsone_context_and_template` to help support new action hooks.
+- added `verify_repo_matches_url` to stop using `.startswith()` to compare urls
+- added `REPO_SCOPE_REGEX` to allow us to find the `repo_scope` in a task's scopes.
+- added `get_repo_scope` to return the `repo_scope` in a task's scopes (or `None`)
+- added a `test/data/cotv3` dir for action hook test data.
+
+### Changed
+- set `cot_version` to 3.
+- set `min_cot_version` to 2.
+- we now require cot artifacts in `verify_docker_image_sha`.
+- we no longer check docker image shas against an allowlist; they either match chain of trust artifact shas, or they're a task type that allows prebuilt docker images. If these are defined in-tree, we trace the request to the tree, so these should be as trustable as the tree in question.
```

Comment:
I'm not sure I understand the motivation for dropping this. While we trust trust the tree, it seems a bunch of the logic in scriptworker is providing an additional level of protection.

Maybe it would be good to have an explicit threat model for what protection scriptworker is providing.